src/vivarium_public_health/disease/state.py

```diff
@@ -458,26 +461,56 @@ def setup(self, builder: Builder) -> None:
         super().setup(builder)
         self.clock = builder.time.clock()
 
-        self.dwell_time = self.get_dwell_time_pipeline(builder)
-        self.disability_weight = self.get_disability_weight_pipeline(builder)
+        builder.value.register_attribute_producer(
+            f"{self.state_id}.dwell_time",
+            source=self.lookup_tables["dwell_time"],
+            component=self,
+            required_resources=get_lookup_columns([self.lookup_tables["dwell_time"]]),
+        )
+        builder.value.register_attribute_producer(
+            f"{self.state_id}.disability_weight",
+            source=self.compute_disability_weight,
+            component=self,
+            required_resources=get_lookup_columns([self.lookup_tables["disability_weight"]])
+            + ["alive", self.model],
+        )
 
         builder.value.register_attribute_modifier(
             "all_causes.disability_weight",
-            modifier=self.disability_weight,
+            modifier=lambda index: self.population_view.get_attributes(
```

Comment:
Isn't this a case where we want to be able to pass a string to a value modifier and have it be recognized as an attribute name?

Reply:
Ugh, yeah. I wonder if that's a simple thing to implement. Seems easier than registering the producer since right now we only support a callable so no need to distinguish things.